### PR TITLE
[WIP] [DOC-10641] Add reference for encode-uri

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-10641] Add reference for encode-uri

- Mark --user and --certs-dir flags as new
- Document all flags
- Add links to pages that refer to the command

TODO: Decide whether to port the reference back through v24.1 (without the new flags) or leave it undocumented in oder versions.

Preview: [src/current/v24.3/cockroach-encode-uri.md](https://deploy-preview-18993--cockroachdb-docs.netlify.app/docs/v24.3/cockroach-encode-uri.html)

### CODE REVIEW

File: src/current/v24.3/cockroach-encode-uri.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: cockroach encode-uri
+summary: Get the connection string for a cluster from its URI.
+toc: true
+docs_area: reference.cli
+---
+
+The `cockroach encode-uri` command returns the `postgres://` connection string for a cluster's `defaultdb` database from the URI of a node.
```
Comment (mdlinville) [inline src/current/v24.3/cockroach-encode-uri.md:8]:
```suggestion
The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster's `defaultdb` database from the URI of a node.
```

File: src/current/v24.3/cockroach-encode-uri.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: cockroach encode-uri
+summary: Get the connection string for a cluster from its URI.
+toc: true
+docs_area: reference.cli
+---
+
+The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster's `defaultdb` database from the URI of a node.
```
Comment (mdlinville) [inline src/current/v24.3/cockroach-encode-uri.md:8]:
```suggestion
The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster and database from the URI of a cluster node.
```

File: src/current/v24.3/cockroach-encode-uri.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+cockroach encode-uri --database {DATABASE} {URI}
+~~~
+
+## Flags
+
+The `encode-uri` command supports the following flags. All flags are optional.
+
+Flag          | Description
+--------------|-----------
+`--inline`    | Whether to inline certificates for [Physical Cluster Replication]({% link {{ page.version.version }}/physical-cluster-replication.md %}). Defaults to `false`.
```
Comment (mdlinville) [inline src/current/v24.3/cockroach-encode-uri.md:38]:
```suggestion
`--inline`    | Whether to inline certificates for [Physical Cluster Replication]({% link {{ page.version.version }}/physical-cluster-replication-overview.md %}). Defaults to `false`.
```

File: src/current/v24.3/cockroach-encode-uri.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster and database from the URI of a cluster node.
+
+## Synopsis
+
+To get the connection string for a cluster's `defaultdb` database, replace `{URI}` with the URI of a cluster node.
+
+~~~ shell
+cockroach encode-uri {URI}
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/v24.3/cockroach-encode-uri.md:15]: Can you verify that the protocol here expects two slashes? It looks like the returned URI contains only a single slash.

File: src/current/v24.3/cockroach-encode-uri.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+---
+title: cockroach encode-uri
+summary: Construct a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide.
+toc: true
+docs_area: reference.cli
+---
+
+The `cockroach encode-uri` command constructs a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide. The returned connection URI is in `postgresql:/` format.
```
Comment (stevendanna) [inline src/current/v24.3/cockroach-encode-uri.md:8]:
```suggestion
The `cockroach encode-uri` command constructs a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide. The returned connection URI is in `postgresql://` format.
```

We only ever want to refer to the scheme as either `postgresql` or `postgresql://`. `postgres:/` will not do what the user thinks.

File: src/current/v24.3/cockroach-encode-uri.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach encode-uri [postgres:/][{USERNAME}[:{PASSWORD}]@]{HOST[:{PORT}]} [{FLAGS}]
+~~~
+
+Only `{HOST}` is required; all other options and [flags](#flags) are optional, as indicated by square brackets.
+
+## Options
+
+- `postgres:/`: Optional protocol specifier if providing a connection URI as input, rather than a `HOST:PORT` combination.
```
Comment (stevendanna) [inline src/current/v24.3/cockroach-encode-uri.md:21]:
```suggestion
- `postgres://`: Optional postgresql URI scheme.
```

**Comment by mdlinville** [comment]
Flipping to draft while some command behavior is under discussion.

**Pull request closed without merging**